devito/data/allocators.py

```diff
@@ -77,7 +77,7 @@ def alloc(self, shape, dtype):
         # cast to 1D array of the specified size
         ctype_1d = ctype * size
         buf = ctypes.cast(c_pointer, ctypes.POINTER(ctype_1d)).contents
-        pointer = np.frombuffer(buf, dtype=dtype).reshape(shape)
+        pointer = np.array(c_pointer, dtype=dtype, copy=False, order='C', subok=True)
```

Comment:
@tjb900 since you had changed these lines, I wonder whether you could take a second look here?

what was the issue before ?

Reply:
Ah I see it now. I worked with numpy v1.14 while developing this. Meanwhile, of course, numpy's API changed and probably that's why @tjb900 made this change. I'll have to give this some more thought.